### ISSUE TITLE
withRowCount() seems to reset its output

### ISSUE THREAD

**Issue by johanroelofsen** [pola-rs/nodejs-polars#15]
#### Have you tried latest version of polars?

- [yes]

#### What version of polars are you using?

tested this with 0.6.0 and 0.5.4

#### What operating system are you using polars on?

Windows 10

#### What node version are you using
node 16.17.0 

#### Describe your bug.

When you chain withRowCount to scanCSV, the row count column seems to reset after a while. 

#### What are the steps to reproduce the behavior?

Use a CSV with sufficient lines (it seems you need at least 1500 lines). Use scanCSV and withRowCount.
When manually initializing the LazyDataFrame, withRowCount seems to work as intended.

```
import pl from "nodejs-polars";
import * as fs from "node:fs";

const data = [...Array(5000).keys()];
fs.writeFileSync("data.csv", data.join("\n"));

const lf1 = await pl.DataFrame(data).lazy().withRowCount().collect();
const lf2 = await pl.scanCSV("data.csv").withRowCount().collect();

console.log(lf1);
console.log(lf2);
```

#### What is the actual behavior?

```
lf1:
┌────────┬──────────┐
│ row_nr ┆ column_0 │
│ ---    ┆ ---      │
│ u32    ┆ f64      │
╞════════╪══════════╡
│ 0      ┆ 0.0      │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 1      ┆ 1.0      │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 2      ┆ 2.0      │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 3      ┆ 3.0      │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ ...    ┆ ...      │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 4996   ┆ 4996.0   │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 4997   ┆ 4997.0   │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 4998   ┆ 4998.0   │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ 4999   ┆ 4999.0   │
└────────┴──────────┘

lf2:
shape: (4999, 2)
┌────────┬──────┐
│ row_nr ┆ 0    │
│ ---    ┆ ---  │
│ u32    ┆ i64  │
╞════════╪══════╡
│ 0      ┆ 1    │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 1      ┆ 2    │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 2      ┆ 3    │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 3      ┆ 4    │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ ...    ┆ ...  │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 1186   ┆ 4996 │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 1187   ┆ 4997 │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 1188   ┆ 4998 │
├╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 1189   ┆ 4999 │
└────────┴──────┘

```

#### What is the expected behavior?

I would expect the withRowCount to not reset halfway

What do you think polars should have done?

**Comment by johanroelofsen** [comment]
This snip might be help illustrating the issue. Here you can see that the withRowCount() seems to "reset", here after 1416

![image](https://user-images.githubusercontent.com/43115095/197476545-ef826aed-eb0a-4b4e-92c6-00d13f9de30d.png)

**Comment by universalmind303** [comment]
this should be fixed in 0.7.1

**Comment by johanroelofsen** [comment]
it is!
